### PR TITLE
AO3-4688 Test deleting a gift exchange

### PR DESCRIPTION
**Pull request by sarken**
https://otwarchive.atlassian.net/browse/AO3-4688

## Purpose

Test the destroy action in gift exchange controller

### CODE REVIEW

File: features/gift_exchanges/challenge_giftexchange.feature [hunk 1]
```diff
@@ -498,3 +498,24 @@ Feature: Gift Exchange Challenge
...
+      And I follow "Delete Challenge"
+    Then I should see "Challenge settings were deleted."
+      And no one should have an assignment for "Bad Gift Exchange"
+      And no one should be signed up for "Bad Gift Exchange"
+      And I should not see "Challenge Settings"
+      And I should not see "Sign-up Summary"
+      And I should not see "Assignments"
+      And I should not see "Requests Summary"
+    When I am on the collections page
+    Then I should see "Bad Gift Exchange"
```
Comment (zz9pzza) [inline features/gift_exchanges/challenge_giftexchange.feature]: New line at the end ?

File: features/step_definitions/challege_gift_exchange_steps.rb [hunk 2]
```diff
@@ -333,3 +333,23 @@
...
+end
+
+Then /^no one should have an assignment for "([^\"]*)"$/ do |challenge|
+  collection = Collection.find_by_title(challenge)
+  %w(myname1 myname2 myname3 myname4).each do |name|
+    user = User.find_by_login("myname1")
+    user.offer_assignments.in_collection(collection).should be_empty
+    user.pinch_hit_assignments.in_collection(collection).should be_empty
+  end
+end
```
Comment (zz9pzza) [inline features/step_definitions/challege_gift_exchange_steps.rb]: Another new line here

File: features/step_definitions/challege_gift_exchange_steps.rb [hunk 3]
```diff
@@ -333,3 +333,23 @@
   step %{I open signups for "Awesome Gift Exchange"}
   step %{everyone has signed up for the gift exchange "Awesome Gift Exchange"}
 end
+
+### Deleting things
+# use the same names as "everyone is signed up"
+
+Then /^no one should be signed up for "([^\"]*)"$/ do |challenge|
+  collection = Collection.find_by_title(challenge)
+  %w(myname1 myname2 myname3 myname4).each do |name|
```
Comment (zz9pzza) [inline features/step_definitions/challege_gift_exchange_steps.rb]: Shouldn't this be a check over all users ?

File: features/step_definitions/challege_gift_exchange_steps.rb [hunk 4]
```diff
@@ -333,3 +333,23 @@
...
+    user = User.find_by_login("myname1")
+    user.challenge_signups.in_collection(collection).should be_empty
+  end
+end
+
+Then /^no one should have an assignment for "([^\"]*)"$/ do |challenge|
+  collection = Collection.find_by_title(challenge)
+  %w(myname1 myname2 myname3 myname4).each do |name|
```
Comment (zz9pzza) [inline features/step_definitions/challege_gift_exchange_steps.rb]: I can't see where name is used ?

Shouldn't be a loop over all users ?

File: features/step_definitions/challege_gift_exchange_steps.rb (hunk 3 above)
Comment (sarken) [inline features/step_definitions/challege_gift_exchange_steps.rb]: No, we just need to check the users who have signed up in "Given everyone has signed up for the gift exchange Bad Gift Exchange":
```
Given /^everyone has signed up for the gift exchange "([^\"]*)"$/ do |challengename|
  step %{I am logged in as "myname1"}
  step %{I sign up for "#{challengename}" with combination A}
  step %{I am logged in as "myname2"}
  step %{I sign up for "#{challengename}" with combination B}
  step %{I am logged in as "myname3"}
  step %{I sign up for "#{challengename}" with combination C}
  step %{I am logged in as "myname4"}
  step %{I sign up for "#{challengename}" with combination D}
end
```

File: features/step_definitions/challege_gift_exchange_steps.rb (hunk 4 above)
Comment (sarken) [inline features/step_definitions/challege_gift_exchange_steps.rb]: Oops, sorry, it should be `user = User.find_by_login(name)`

File: features/step_definitions/challege_gift_exchange_steps.rb (hunk 3 above)
Comment (zz9pzza) [inline features/step_definitions/challege_gift_exchange_steps.rb]: I think it should check all users. It hurts the eye to see all users then a literal list :(

Comment (sarken) [inline features/step_definitions/challege_gift_exchange_steps.rb]: How about if I change it to `everyone's assignments for "([^\"]*)" should be deleted` and `everyone's sign ups for "([^\"]*)" should be deleted`? I don't want to rename the initial (and equally misleading) `everyone has signed up for the gift exchange "([^\"]*)"` step, but I also don't want to check more than we need to.

**Comment by sarken** [comment]
Closing while I add a few more tests

**Pull request closed without merging**